### PR TITLE
docs: update code in tutorial and dcutr docs

### PR DESCRIPTION
**Pull request by dvorakchen**
## Description

When we moved our examples, the hole-punching docs were not updated for the new names of the binaries. We adjust them such that the tutorial works again.

## Notes & open questions

I have been greatly impeded in my learning process due to errors in documentation

The code in tutorial and example dcutr not work

In tutorial:
line 57, 'cargo build --example relay_v2 -p libp2p-relay', will ouput: 
'error: no example target named `relay_v2`.'
Actually we have not example named 'relay_v2', I thought maybe renamed 'relay-server-example'
Then updated to 
```
cd ./examples/relay-server
cargo build
```

The same applies to the 'client' on line 126, it will output: `error: no example target named `client`.`
Cause has not example named 'client', maybe renamed to 'dcutr'

In examples/dcutr:
line 21 would panic, due to the `--relay-address` miss `p2p` component
Related: [dcutr in examples executes incorrectly #4385](https://github.com/libp2p/rust-libp2p/discussions/4385)

And parameter `--relay-address /ip4/127.0.0.1/tcp/12345` could not work, It should be a public relay address, cannot be a loopback address. It would panic at 'examples/dcutr/src/main' line 203

So I changed that loopback address to '/ip4/$RELAY_IP/tcp/$PORT/p2p/$RELAY_PEERID', The relay address is provided by the user

Then they work correctly.

## Change checklist

- [Y] I have performed a self-review of my own code
- [Y] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: libp2p/src/tutorials/hole_punching.rs [hunk 1]
```diff
@@ -54,16 +54,17 @@
 //!
 //! ``` bash
 //! ## Inside the rust-libp2p repository.
-//! cargo build --example relay_v2 -p libp2p-relay
+//! cd ./examples/relay-server
+//! cargo build
```
Comment (thomaseizinger) [inline libp2p/src/tutorials/hole_punching.rs]: I think we can simplify this a bit so users don't have to change directories :)

```suggestion
//! cargo build --bin relay-server-example
```

File: libp2p/src/tutorials/hole_punching.rs [hunk 2]
```diff
@@ -122,16 +123,17 @@
 //!
 //! ``` bash
 //! ## Inside the rust-libp2p repository.
-//! cargo build --example client -p libp2p-dcutr
+//! cd ./examples/relay-server
+//! cargo build
```
Comment (thomaseizinger) [inline libp2p/src/tutorials/hole_punching.rs]: Seems to be a copy-paste error here. I'd also recommend that we specify the binary directly:

```suggestion
//! cargo build --bin dcutr
```

File: libp2p/src/tutorials/hole_punching.rs [hunk 3]
```diff
@@ -122,16 +123,17 @@
...
 //!
-//! You can find the binary at `target/debug/examples/client`. In case you built it locally, copy
+//! You can find the binary at `target/debug/dcutr`. In case you built it locally, copy
 //! it to your listening client machine.
 //!
 //! On the listening client machine:
 //!
 //! ``` bash
-//! RUST_LOG=info ./client --secret-key-seed 1 --mode listen --relay-address /ip4/$RELAY_SERVER_IP/tcp/4001/p2p/12D3KooWDpJ7As7BWAwRMfu1VU2WCqNjvq387JEYKDBj4kx6nXTN
+//! RUST_LOG=info ./dcutr --secret-key-seed 1 --mode listen --relay-address /ip4/$RELAY_SERVER_IP/tcp/4001/p2p/12D3KooWDpJ7As7BWAwRMfu1VU2WCqNjvq387JEYKDBj4kx6nXTN
```
Comment (thomaseizinger) [inline libp2p/src/tutorials/hole_punching.rs:134]: I can't comment on it but I think line 158 also needs to be updated! :)

**Review by thomaseizinger — COMMENTED** [review]
Thank you for these fixes! :)

A few, small comments!

**Comment by thomaseizinger** [comment]
@dvorakchen FYI: I've updated the PR description in preparation for merging this to make a nice commit message.

File: libp2p/src/tutorials/hole_punching.rs (hunk 3 above)
Comment (dvorakchen) [reply libp2p/src/tutorials/hole_punching.rs:134]: Ok, I updated line 158. Thank a lot

**Review by thomaseizinger — DISMISSED** [review]

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**